mcs/class/corlib/System.IO/LogcatTextWriter.cs

```diff
@@ -48,22 +48,52 @@ public override void WriteLine ()
 			var o = line.ToString ();
 			line.Clear ();
 
+			Log (o);
+			stdout.WriteLine (o);
+		}
+
+		void Log (string message)
+		{
+			const int buffer_size = 512;
+
 			unsafe {
-				fixed (byte *b_appname = appname)
-				fixed (byte *b_message = Encoding.UTF8.GetBytes(o)) {
-					Log (b_appname, 1 << 5 /* G_LOG_LEVEL_MESSAGE */, b_message);
+				if (message.Length < buffer_size) {
```

Comment:
This is wrong. You don't want `message.Length < buffer_size`, as `message` may contain e.g. emoji, which will take more than one byte to convert.

Instead, you want `if (Encoding.UTF8.GetByteCount (message) < buffer_size)`.